src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowConnector.java

```diff
@@ -0,0 +1,167 @@
+import io.cdap.cdap.etl.api.connector.BrowseEntity;
+import io.cdap.cdap.etl.api.connector.BrowseEntityPropertyValue;
+import io.cdap.cdap.etl.api.connector.BrowseRequest;
+import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.cdap.etl.api.connector.ConnectorContext;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.plugin.common.ConfigUtil;
+import io.cdap.plugin.servicenow.apiclient.ServiceNowTableAPIClientImpl;
+import io.cdap.plugin.servicenow.apiclient.ServiceNowTableAPIRequestBuilder;
+import io.cdap.plugin.servicenow.restapi.RestAPIResponse;
+import io.cdap.plugin.servicenow.source.ServiceNowInputFormat;
+import io.cdap.plugin.servicenow.util.ServiceNowConstants;
+import io.cdap.plugin.servicenow.util.ServiceNowTableInfo;
+import io.cdap.plugin.servicenow.util.SourceQueryMode;
+import io.cdap.plugin.servicenow.util.SourceValueType;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+
+
+/**
+ * ServiceNow Connector Plugin
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(ServiceNowConnector.NAME)
+@Description("Connection to access data in Servicenow tables.")
+public class ServiceNowConnector implements Connector {
+  public static final String NAME = "ServiceNow";
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowConnector.class);
+  private static final String OBJECT_TABLE_LIST = "sys_db_object";
+  private static final String ENTITY_TYPE_TABLE = "table";
+  private static final String LABEL_NAME = "label";
+  private static final String OAUTH_URL_TEMPLATE = "%s/oauth_token.do";
+
+  private final ServiceNowConnectorConfig config;
+
+  ServiceNowConnector(ServiceNowConnectorConfig config) {
+    this.config = config;
+
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    int count = 0;
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+    collector.getOrThrowException();
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    TableList tables = listTables();
+    Table[] table = tables.getResult();
+    for (int i = 0; i < table.length; i++) {
+      String name = table[i].getName();
+      String label = table[i].getLabel();
```

Comment:
We can remove name and label variable

Reply:
It is the standard implementation which has been done in other connectors as well. Do u want me to remove it?